forest/oak/base.py

```diff
@@ -628,15 +640,19 @@ def run(study_folder: str, output_folder: str, tz_str: Optional[str] = None,
                 ]
                 # transform t to full hours
                 t_series = pd.Series(t_datetime)
-                t_hours_pd = t_series.dt.floor('H')
+                if frequency == Frequency.MINUTELY:
+                    t_hours_pd = t_series.dt.floor('T')
+                else:
+                    t_hours_pd = t_series.dt.floor('H')
+
                 # convert t_hours to correct timezone
                 t_hours_pd = t_hours_pd.dt.tz_localize(
                     from_zone
                 ).dt.tz_convert(to_zone)
 
                 run_hourly(
-                    t_hours_pd, days_hourly, cadence_bout, steps_hourly,
-                    walkingtime_hourly, cadence_hourly, frequency
+                    t_hours_pd, t_ind_pydate, cadence_bout, steps_hourly,
+                    walkingtime_hourly, cadence_hourly, freq_value
```

Comment:
Local variables `t_ind_pydate`, `steps_hourly`, `walkingtime_hourly`, and `cadence_hourly` might be referenced before assignment.